src/utils/analytics.ts

```diff
@@ -0,0 +1,23 @@
+import type { PlaylistItem, Source } from '#types/playlist';
+
+export const attachAnalyticsParams = (item: PlaylistItem, isJwIntegration: boolean, userId?: string, profileId?: string) => {
+  const { sources, mediaid } = item;
+
+  return sources.map((source: Source) => {
+    const url = new URL(source.file);
```

Comment:
Just to be safe with case sensitivity:
```suggestion
    const url = new URL(source.file.toLowerCase());
```